### PR TITLE
Update: Ignore end of function in newline-after-var (fixes #3682)

### LINKED ISSUE #3682: ignore end of function, or add parameter for ignore in rule newline-after-var
```json
{
"newline-after-var": [
            2,
            "always"
        ]
}
```
```javascript
function example() {
    var anonymous = function () {
        console.log('anonymous function expression');
    };
}
```

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/26503548-ignore-end-of-function-or-add-parameter-for-ignore-in-rule-newline-after-var?utm_campaign=plugin&utm_content=tracker%2F282608&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F282608&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by alberto**

### CODE REVIEW

File: lib/rules/newline-after-var.js [hunk 1]
```diff
@@ -114,6 +114,12 @@ module.exports = function(context) {
             return;
         }
 
+        // Ignore if it is last statement in a function
+        if (node.parent.parent && node.parent.parent.type === "FunctionDeclaration" &&
```
Comment (nzakas) [inline lib/rules/newline-after-var.js]: This check isnt quite right. First, we need to check for all function forms (FunctionDeclaration, FunctionExpression, ArrowFunctionExpression).

Next, you can check that the last element in the body is equal to the VariableDeclaration.

Comment (alberto) [inline lib/rules/newline-after-var.js]: @nzakas Is it ok now?

File: lib/rules/newline-after-var.js [hunk 2]
```diff
@@ -64,6 +64,27 @@ module.exports = function(context) {
...
+    }
+
+    /**
+     * Determine if provided node is the last of his parent
+     * @private
+     * @param {ASTNode} node - node to test
+     * @returns {boolean} True if `node` is last of his parent
+     */
+    function isLastNode(node) {
+        return node.parent.body.pop() === node;
```
Comment (nzakas) [inline lib/rules/newline-after-var.js]: This changes what's in body, so it will break other rules. You should just check the node at node.body.length - 1.

Comment (alberto) [inline lib/rules/newline-after-var.js]: Ouch

Comment (alberto) [inline lib/rules/newline-after-var.js]: Fixed (hopefully)

**Comment by nzakas** [comment]
Lgtm

**Pull request merged**